sixtyfps_compiler/object_tree.rs

```diff
@@ -1283,12 +1283,26 @@ fn animation_element_from_node(
     } else {
         let mut anim_element =
             Element { id: "".into(), base_type: anim_type, node: None, ..Default::default() };
+
         anim_element.parse_bindings(
             anim.Binding().filter_map(|b| {
                 Some((b.child_token(SyntaxKind::Identifier)?, b.BindingExpression().into()))
             }),
             diag,
         );
+
+        // Apply default property values on top:
+        if let Type::Builtin(builtin_base) = &anim_element.base_type {
+            for (prop, info) in &builtin_base.properties {
+                if let Some(expr) = &info.default_value {
+                    anim_element
+                        .bindings
+                        .entry(prop.clone())
+                        .or_insert_with(|| RefCell::new(expr.clone().into()));
+                }
+            }
+        }
```

Comment:
Just a thought: Since Element::from_node has "similar" code, maybe it's worth sharing this in a helper function? Like this:

```rust
fn apply_default_values(bindings: &mut BindingsMap, base_type: &Type) {
    // ...
}
```

and `Element::from_node` could call it with `(&mut anim_element.bindings, &anim_element.base_type)`

Reply:
Good point, done.